src/primitives/triangle/mod.rs

```diff
@@ -144,39 +135,38 @@ impl ContainsPoint for Triangle {
 
 impl Dimensions for Triangle {
     fn bounding_box(&self) -> Rectangle {
-        let x_min = min(min(self.p1.x, self.p2.x), self.p3.x);
-        let y_min = min(min(self.p1.y, self.p2.y), self.p3.y);
+        let [p1, p2, p3] = self.vertices;
+
+        let x_min = min(min(p1.x, p2.x), p3.x);
+        let y_min = min(min(p1.y, p2.y), p3.y);
 
-        let x_max = max(max(self.p1.x, self.p2.x), self.p3.x);
-        let y_max = max(max(self.p1.y, self.p2.y), self.p3.y);
+        let x_max = max(max(p1.x, p2.x), p3.x);
+        let y_max = max(max(p1.y, p2.y), p3.y);
 
         Rectangle::with_corners(Point::new(x_min, y_min), Point::new(x_max, y_max))
     }
 }
 
 impl Triangle {
-    /// Create a new triangle with a given style
-    pub const fn new(p1: Point, p2: Point, p3: Point) -> Self {
-        Triangle { p1, p2, p3 }
+    /// Create a new triangle with the given vertices.
+    pub const fn new(vertex1: Point, vertex2: Point, vertex3: Point) -> Self {
+        Triangle {
+            vertices: [vertex1, vertex2, vertex3],
+        }
     }
 
-    /// Creates a new triangle from an array of points.
+    /// Creates a new triangle from a [`Point`] slice.
     ///
-    /// This supports both [`Point`]s, as well as anything that implements `Into<Point>` like
-    /// `(i32, i32)`.
+    /// # Panics
+    ///
+    /// This method will panic if the given slice is not exactly 3 items long.
     ///
     /// [`Point`]: ../../geometry/struct.Point.html
-    pub fn from_points<P, I>(points: P) -> Self
-    where
-        I: Into<Point> + Copy,
-        P: Borrow<[I; 3]>,
-    {
-        let points = points.borrow();
-
-        Triangle {
-            p1: points[0].into(),
-            p2: points[1].into(),
-            p3: points[2].into(),
+    // MSRV: Consider subslice patterns for rust >= 1.42.0
```

Comment:
We don't need subslice patterns anymore, now that we require exactly 3 elements.
```suggestion
```